misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java

```diff
@@ -564,24 +564,35 @@ private void executeFunction(Test test, TestSuite suite, String packageName, Cla
             } else {
                 if (valueSets instanceof BMap) {
                     // Handle map data sets
-                    List<String> keyValues = getKeyValues((BMap) valueSets);
-                    Class<?>[] argTypes = extractArgumentTypes((BMap) valueSets);
-                    List<Object[]> argList = extractArguments((BMap) valueSets);
-                    int i = 0;
-                    for (Object[] arg : argList) {
-                        invokeDataDrivenTest(suite, test.getTestName(), escapeSpecialCharacters(keyValues.get(i)),
-                                classLoader, scheduler, shouldSkip, packageName, arg, argTypes, failedOrSkippedTests);
-                        i++;
+                    if (((BMap) valueSets).isEmpty()) {
+                        computeFunctionResult(test.getTestName(), packageName, shouldSkip, failedOrSkippedTests,
+                                new Error("The provided data set is empty."));
+                    } else {
+                        List<String> keyValues = getKeyValues((BMap) valueSets);
+                        Class<?>[] argTypes = extractArgumentTypes((BMap) valueSets);
+                        List<Object[]> argList = extractArguments((BMap) valueSets);
+                        int i = 0;
+                        for (Object[] arg : argList) {
+                            invokeDataDrivenTest(suite, test.getTestName(), escapeSpecialCharacters(keyValues.get(i)),
+                                    classLoader, scheduler, shouldSkip, packageName, arg, argTypes,
+                                    failedOrSkippedTests);
+                            i++;
+                        }
```

Comment:
We can avoid the temp variable `i` by using an optimised indexed for loop instead. WDYT?

Reply:
Fixed with 92b98e5037942465b58076ce069597e5f12ae983